### PR TITLE
Update deployment pipeline for challenge based worker

### PR DESCRIPTION
**Pull request by RishabhJain2018**

### CODE REVIEW

File: scripts/deployment/deploy.sh [hunk 1]
```diff
@@ -47,16 +47,39 @@ case $opt in
             if [ ${env} == "production" ]; then
                 echo "Pulling ssl certificates and nginx configuration..."
                 aws s3 cp s3://cloudcv-secrets/evalai/${env}/ssl/ ./ssl/ --recursive
-                aws s3 cp s3://cloudcv-secrets/evalai/${env}/nginx_${env}.conf ./docker/prod/nodejs/nginx_${env}.conf
+                aws s3 cp s3://cloudcv-secrets/evalai/${env}/nginx_${env}.conf ./docker/prod/nodejs/nginx_${env}.conf --recursive
```
[the remark this answers is not in the document]
Comment (RishabhJain2018) [reply scripts/deployment/deploy.sh]: We need it. I am not sure of the reason though why we need it but I tried it last time and it failed without it.

File: scripts/deployment/deploy.sh [hunk 2]
```diff
@@ -47,16 +47,39 @@ case $opt in
...
+            ;;
+        deploy-nodejs)
+            echo "Deploying nodejs docker container..."
+            docker-compose -f docker-compose-${env}.yml up -d nodejs
+            echo "Completed deploy operation."
+            ;;
+        deploy-worker)
+            token=${3}
+            if [ ${env} == "staging" ]; then
+                echo "Pulling queue names for staging server challenges..."
```
[the remark this answers is not in the document]
Comment (RishabhJain2018) [reply scripts/deployment/deploy.sh]: Updated it.

File: scripts/deployment/deploy.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (deshraj) [reply scripts/deployment/deploy.sh]: I am quite certain that we don't need it and it should just work fine.

File: scripts/deployment/deploy.sh [hunk 3]
```diff
@@ -47,16 +47,37 @@ case $opt in
...
+            docker-compose -f docker-compose-${env}.yml up -d nodejs
+            echo "Completed deploy operation."
+            ;;
+        deploy-worker)
+            token=${3}
+            echo "Pulling queue names for $env server challenges..."
+            if [ ${env} == "staging" ]; then
+                queue_names=$(curl -L -X GET -H "Authorization: Token $token" http://staging.evalai.cloudcv.org/api/challenges/get_broker_urls) | tr -d '[],'
+            fi
+            if [ ${env} == "production" ]; then
```
Comment (deshraj) [inline scripts/deployment/deploy.sh]: Shouldn't this be `elif` instead of `if`

**Pull request merged**